### PR TITLE
[code-quality] fix clang-tidy md5 and hmac_md5

### PR DESCRIPTION
**Pull request by tomaszduda23**
# What does this implement/fix?

```
### File /esphome/.temp/all-include.cpp

/esphome/esphome/components/md5/md5.h:63:3: error: unknown type name 'MD5_CTX_TYPE' [clang-diagnostic-error]
  MD5_CTX_TYPE ctx_{};
  ^
```

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by dwmw2 — APPROVED** [review]
Looks reasonable; thanks.

File: esphome/components/md5/md5.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include <cstdio>
```
Comment (guillempages) [inline esphome/components/md5/md5.cpp:1]: Could/should the `<cstdio>` and `<cstring>` includes be moved inside the `#ifdef` as well? Or does than the linter complain?

Comment (tomaszduda23) [reply esphome/components/md5/md5.cpp:1]: it does not matter. I updated.

Comment (tomaszduda23) [reply esphome/components/md5/md5.cpp:1]: well. I rolled it back since there is compilation error.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**